packages/typespec-azure-portal-core/src/decorators.ts

```diff
@@ -0,0 +1,253 @@
+  return program.stateMap(PortalCoreKeys.browse).get(target);
+}
+
+export function getBrowseArgQuery(program: Program, target: Type) {
+  const browse = getBrowse(program, target);
+  return browse.properties.get("argQuery");
+}
+
+export function isARMResource(
+  program: Program,
+  target: Model,
+  decoratorName: "browse" | "about" | "marketplaceOffer"
+) {
+  if (
+    getArmResourceKind(target) !== ("Tracked" as ArmResourceKind) &&
+    getArmResourceKind(target) !== ("Proxy" as ArmResourceKind)
+  ) {
+    reportDiagnostic(program, {
+      code: "invalidUsageDecorator",
+      messageId: decoratorName,
+      target,
+    });
+  }
+}
+/**
+ * This is a About decorator that will be used to define icon, keywords and learnMoreDocs.
+ * @param target The model that is being decorated.
+ * @param options AboutOptions of the property.
+ */
+export function $about(context: DecoratorContext, target: Model, options: AboutOptions) {
+  const { program } = context;
+  validateDecoratorUniqueOnNode(context, target, $about);
+  isARMResource(program, target, "about");
+  if (options && (options as Model).properties) {
+    const icon = (options as Model).properties.get("icon");
+    if (icon) {
+      if (icon.type.kind === "Model") {
+        const sourceLocation = getSourceLocation(target);
+        const dirPath =
+          sourceLocation && sourceLocation.file.path && getDirectoryPath(sourceLocation.file.path);
+        let iconPath = icon.type && (icon.type as Model).properties.get("filePath");
+        const iconPathValue = iconPath && iconPath.type && (iconPath.type as StringLiteral).value;
+        let filePath = resolvePath(dirPath, iconPathValue); //if given path is fullpath, it will return the fullPath
+        if (filePath && iconPath && iconPathValue) {
+          filePath = normalizePath(filePath);
+          if (!fs.existsSync(filePath)) {
+            reportDiagnostic(program, {
+              code: "fileNotFound",
+              messageId: "aboutIcon",
+              target,
+            });
+          }
+          (iconPath.type as StringLiteral).value = filePath;
+        }
+      }
+    }
+  }
+  program.stateMap(PortalCoreKeys.about).set(target, options);
+}
+
+export function getAbout(program: Program, target: Type) {
+  return program.stateMap(PortalCoreKeys.about).get(target);
+}
+
+export function getAboutDisplayName(program: Program, target: Type) {
+  const about = getAbout(program, target);
+  return about.properties.get("displayName");
+}
+
+export function getAboutKeywords(program: Program, target: Type) {
+  const about = getAbout(program, target);
+  return about.properties.get("keywords");
+}
+
+export function getAboutKeywordsItemsArray(program: Program, target: Type) {
+  const keywords = getAboutKeywords(program, target);
+  return keywords.type.values
+    .filter((value: Type) => value.kind === "String")
+    .map((value: Type) => (<StringLiteral>value).value);
```

Comment:
tip: I recommend using new-style `value as StringLiteral` rather than old-style `<StringLiteral>value`.

Reply:
fixed :) thank you!